core/src/multi_proof_verification.rs

```diff
@@ -0,0 +1,636 @@
+use crate::{
+    multi_proof::{MultiProof, SubPathProof},
+    proof::{
+        hash_path, shared_bits, PathProofTerminal, PathProofVerificationError, VerifiedPathProof,
+    },
+    trie::{InternalData, KeyPath, LeafData, Node, NodeHasher, NodeHasherExt, TERMINATOR},
+};
+use alloc::collections::BTreeMap;
+use bitvec::{order::Msb0, vec::BitVec, view::BitView};
+
+/// A verifiable proof of multiple path through the trie.
+#[derive(Debug, Clone)]
+pub struct VerifiableMultiProof<H: NodeHasher> {
+    nodes: BTreeMap<BitVec<u8, Msb0>, Node>,
+    terminals: BTreeMap<KeyPath, Option<LeafData>>,
+    root: Node,
+    _phantom: core::marker::PhantomData<H>,
+}
+
+struct SubTrieRoot {
+    root: Node,
+    key_path: BitVec<u8, Msb0>,
+}
+
+#[derive(Debug, Clone)]
+pub enum MultiProofVerificationError {
```

Comment:
docs on public items.

also, this module uses `PathProofVerificationError` and `PathProofVerificationError`. You should define a single new error type and use that instead of mixing the two.